From 2977de4e4b9110df04369d22f0fc5349d4950b8e Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Wed, 29 Aug 2018 15:14:32 +0800
Subject: [PATCH] refpolicy: fix policy for syslog-ng

syslog-ng in WRLinux intends to create sockfiles and pid file in 
${localstatedir}/run/${PN}, add a allow rule for this.

Default syslog-ng config appends (warn..emerg) messages to /dev/tty10
and /dev/xconsole, also fix this.

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Xin Ouyang <Xin.Ouyang@windriver.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/logging.fc | 1 +
 policy/modules/system/logging.te | 8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/policy/modules/system/logging.fc b/policy/modules/system/logging.fc
index cae0cb0..ec3beb7 100644
--- a/policy/modules/system/logging.fc
+++ b/policy/modules/system/logging.fc
@@ -7,6 +7,7 @@
 /etc/rc\.d/init\.d/auditd --	gen_context(system_u:object_r:auditd_initrc_exec_t,s0)
 /etc/rc\.d/init\.d/rsyslog --	gen_context(system_u:object_r:syslogd_initrc_exec_t,s0)
 /etc/rc\.d/init\.d/syslog\.sysklogd --	gen_context(system_u:object_r:syslogd_initrc_exec_t,s0)
+/etc/rc\.d/init\.d/syslog\.syslog-ng --	gen_context(system_u:object_r:syslogd_initrc_exec_t,s0)
 
 /sbin/audispd		--	gen_context(system_u:object_r:audisp_exec_t,s0)
 /sbin/audisp-remote	--	gen_context(system_u:object_r:audisp_remote_exec_t,s0)
diff --git a/policy/modules/system/logging.te b/policy/modules/system/logging.te
index a5705f1..d617a6e 100644
--- a/policy/modules/system/logging.te
+++ b/policy/modules/system/logging.te
@@ -419,6 +419,7 @@ files_search_var_lib(syslogd_t)
 
 # manage pid file
 manage_files_pattern(syslogd_t, syslogd_var_run_t, syslogd_var_run_t)
+manage_sock_files_pattern(syslogd_t, syslogd_var_run_t, syslogd_var_run_t)
 files_pid_filetrans(syslogd_t, syslogd_var_run_t, file)
 
 kernel_read_system_state(syslogd_t)
@@ -495,6 +496,12 @@ init_read_utmp(syslogd_t)
 init_dontaudit_write_utmp(syslogd_t)
 term_write_all_ttys(syslogd_t)
 
+# default syslog-ng config appends (warn..emerg) messages to /dev/tty10
+term_append_unallocated_ttys(syslogd_t)
+term_dontaudit_setattr_unallocated_ttys(syslogd_t)
+term_use_all_ttys(syslogd_t)
+term_use_all_ptys(syslogd_t)
+
 auth_use_nsswitch(syslogd_t)
 
 init_use_fds(syslogd_t)
@@ -577,6 +584,7 @@ optional_policy(`
 optional_policy(`
 	# log to the xconsole
 	xserver_rw_console(syslogd_t)
+	xserver_setattr_console_pipes(syslogd_t)
 ')
 
 
-- 
2.7.4

